File: sample_code/sample.cpp
#include <bits/stdc++.h>
using namespace std;

int main(){
    int N,M;
    cin>>N>>M;
    vector<string> A(N);
    for(int i=0; i<N; i++){
        cin>>A[i];
    }
    vector<pair<int,vector<pair<int,int>>>> bomb(M);
    for(int i=0; i<M; i++){
        int C,L;
        bomb[i].first=C;
        for(int j=0; j<L; j++){
            int a,b;
            cin>>a>>b;
            bomb[i].second.push_back(make_pair(a,b));
        }
    }
    int x=-1,y=-1;
    for(int i=0; i<N; i++){
        for(int j=0; j<N; j++){
            if(A[i][j]=='@'){
                x=i;
                y=j;
                break;
            }
        }
        if(x!=-1)break;
    }
    vector<pair<int,string>> ans;
    for(int i=0; i<x; i++){
        ans.push_back(make_pair(1,"D"));
    }
    for(int i=0; i<y; i++){
        ans.push_back(make_pair(1,"R"));
    }
    for(int i=0; i<N*N; i++){
        ans.push_back(make_pair(2,to_string(1)));
    }
    for(int i=0; i<x; i++){
        ans.push_back(make_pair(1,"U"));
    }
    for(int i=0; i<y; i++){
        ans.push_back(make_pair(1,"L"));
    }
    for(int i=0; i<N; i++){
        for(int j=0; j<N; j++){
            ans.push_back(make_pair(3,to_string(1)));
            if(i==N-1&&j==N-1)break;
            else if(j==N-1)ans.push_back(make_pair(1,"D"));
            else if(i%2==0)ans.push_back(make_pair(1,"R"));
            else ans.push_back(make_pair(1,"L"));
        }
    }
    cout<<ans.size()<<endl;
    for(pair<int,string> i:ans){
        cout<<i.first<<" "<<i.second<<endl;
    }
}

File: sample_code/sample.py
N,M=map(int,input().split())
A=[]
for i in range(N):
    A.append(input())
bomb=[]
for i in range(M):
    C,L=map(int,input().split())
    bomb.append([C,[]])
    for j in range(L):
        bomb[-1][-1].append(tuple(map(int,input().split())))
x=-1
y=-1
for i in range(N):
    for j in range(N):
        if A[i][j]=="@":
            x=i
            y=j
            break
    if x!=-1:
        break
ans=[]
for i in range(x):
    ans.append([1,"D"])
for i in range(y):
    ans.append([1,"R"])
for i in range(N*N):
    ans.append([2,1])
for i in range(x):
    ans.append([1,"U"])
for i in range(y):
    ans.append([1,"L"])
for i in range(N):
    for j in range(N):
        ans.append([3,1])
        if j==N-1 and i==N-1:
        	break
        if j==N-1:
            ans.append([1,"D"])
        elif i%2==0:
            ans.append([1,"R"])
        else:
            ans.append([1,"L"])
print(len(ans))
for i in ans:
    print(*i)
